semgrep/semgrep/formatter/text.py

```diff
@@ -413,8 +427,60 @@ def format(
             else iter([])
         )
 
-        matches_output = "\n".join((*output, *timing_output))
-        if not matches_output:
-            return ""
+        findings_output = []
+        if reachable or unreachable:
+            findings_output.append(
+                f"\n{with_color(Colors.foreground, 'SCA Summary')}: {len(reachable)} {with_color(Colors.red,'Reachable')} findings, {len(unreachable)} {with_color(Colors.yellow,'Unreachable')} findings\n"
+            )
+
+        if reachable:
+            reachable_output = self._build_text_output(
+                reachable,
+                extra.get("color_output", False),
+                extra["per_finding_max_lines_limit"],
+                extra["per_line_max_chars_limit"],
+            )
+
+            findings_output.append(
+                f"\n{with_color(Colors.red, 'Reachable SCA Findings:')}\n"
+                + "\n".join(reachable_output)
+            )
+
+        if unreachable:
+            unreachable_output = self._build_text_output(
+                unreachable,
+                extra.get("color_output", False),
+                extra["per_finding_max_lines_limit"],
+                extra["per_line_max_chars_limit"],
+            )
+
+            findings_output.append(
+                f"\n{with_color(Colors.yellow, 'Unreachable SCA Findings:')}\n"
+                + "\n".join(unreachable_output)
+            )
 
-        return "\nFindings:\n" + matches_output
+        if (reachable or unreachable) and first_party:
+            first_party_output = self._build_text_output(
+                first_party,
+                extra.get("color_output", False),
+                extra["per_finding_max_lines_limit"],
+                extra["per_line_max_chars_limit"],
+            )
+            findings_output.append(
+                "\nFirst-Party Findings:\n" + "\n".join(first_party_output)
```

Comment:
I'm not fully sold on "First-Party Findings" Does just "Findings" cause confusion?

Reply:
I thought it seemed slightly weird since "Findings" sounds like it ought to include _all_ findings to me. But I don't feel very strongly, happy to change it back

Reply:
I don't feel strongly either. Let's keep for now and see if it needs changing later